glue_runner.py

```diff
@@ -0,0 +1,276 @@
+    "-r",
+    "--region",
+    help="AWS region to setup (default: us-east-1).",
+    default="us-east-1",
+    type=str,
+)
+parser.add_argument(
+    "-i",
+    "--glue-iam-role",
+    help="Name of IAM role onboarded for use with Glue.",
+    type=str,
+)
+parser.add_argument(
+    "-w",
+    "--glue-max-workers",
+    help="Maximum number of glue workers (default: 25).",
+    default=25,
+    type=int,
+)
+parser.add_argument(
+    "-n",
+    "--glue-job-name",
+    help="Name of the Glue Job to create/update (default: deltacat-example-runner-$USER).",
+    type=str,
+)
+args_mutex = parser.add_mutually_exclusive_group()
+args_mutex.add_argument(
+    "-d",
+    "--deploy-local-deltacat",
+    help="Builds and deploys your local workspace code for DeltaCAT "
+    "(reusable for 7 days).",
+    action="store_true",
+)
+args_mutex.add_argument(
+    "-v",
+    "--deltacat-version",
+    help="DeltaCAT version to install from PyPi.",
+    type=str,
+)
+args_mutex.add_argument(
+    "-u",
+    "--deltacat-url",
+    help="Installs DeltaCAT from the given URL (via pip).",
+    type=str,
+)
+
+ROOT_DIR = os.path.dirname(__file__)
+
+
+if __name__ == "__main__":
+    import subprocess
+    import boto3
+    import json
+    from botocore.exceptions import ClientError
+
+    this_script_dir = os.path.realpath(ROOT_DIR)
+    print(f"Script Parent Directory: {this_script_dir}")
+
+    # set the stage to the current user retrieved from system environment vars
+    stage = os.environ["USER"]
+    print(f"Developer Stage: {stage}")
+    args = parser.parse_args()
+
+    # get any custom version or uri of deltacat to install
+    dc_version = args.deltacat_version
+    dc_url = args.deltacat_url
+
+    # run a build & deploy if needed
+    deploy_dc = args.deploy_local_deltacat
+    if deploy_dc:
+        # delegate build and deploy to S3 to shell script
+        cmd = [f"{this_script_dir}/s3-build-and-deploy.sh"]
+        print(f"Running Command: {cmd}")
+        proc = subprocess.Popen(
+            cmd,
+            stdout=subprocess.PIPE,
+            # redirect stderr to stdout
+            stderr=subprocess.STDOUT,
+            # set working directory to this script's directory
+            cwd=f"{this_script_dir}",
+        )
+        while True:
+            line = proc.stdout.readline().decode("utf-8")
+            print(f"{line}")
+            if not line and proc.poll() is not None:
+                break
+            line = line.strip()
+            if line.find("DELTACAT_PRESIGNED_URL=") != -1:
+                dc_url = line.split("=", 1)[1]
+        returncode = proc.poll()
+        if returncode != 0:
+            raise RuntimeError(f"Command {cmd} Failed. Exit Code: {returncode}")
+
+    s3_client = boto3.client("s3")
+
+    # upload the example runner script to S3
+    bucket_name = f"deltacat-glue-scripts-{stage}"
+    print(f"Ensuring Glue Job Script Bucket exists: {bucket_name}")
+    s3_client.create_bucket(Bucket=bucket_name)
```

Comment:
`LocationConstraint` must be added to create buckets in regions other than `us-east-1`. Ref: https://awscli.amazonaws.com/v2/documentation/api/latest/reference/s3api/create-bucket.html

Reply:
Good catch - added!